From a821dc9961d457c086fffcc16a911cb6f9f8659a Mon Sep 17 00:00:00 2001
From: xiliang <xiliang@redhat.com>
Date: Sun, 18 Jun 2017 00:33:28 +0800
Subject: [PATCH] As xen pv guest can access cpuid from Intel CPUs started
 IvyBridge onwards have CPUID Faulting, added one more check in virt-what.

---
 virt-what.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/virt-what.in b/virt-what.in
index a05e0db..247348e 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -1,6 +1,6 @@
 #!/bin/sh -
 # @configure_input@
-# Copyright (C) 2008-2015 Red Hat Inc.
+# Copyright (C) 2008-2017 Red Hat Inc.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -223,7 +223,8 @@ fi
 
 # Check for Xen.
 
-if [ "$cpuid" = "XenVMMXenVMM" ]; then
+if [ "$cpuid" = "XenVMMXenVMM" ] &&
+    ! echo "$dmi" | grep -q 'No SMBIOS nor DMI entry point found, sorry'; then
     echo xen; echo xen-hvm
     # Check for AWS
     if echo "$dmi" | grep -q 'Version: [0-9]\.[0-9]\.amazon'; then
-- 
2.15.0

